### DISCUSSION TITLE
Querying the databse

### DISCUSSION THREAD

**Discussion by DdHms** [immich-app/immich#12459] [Q&A]
Hi, 
I would like to query the database outside immich app for all images of certain faces. For example:
give me the paths to all images that contain [Face1] or/and [Face2] (assumming those faces have names).
Could anyone direct me to a place where I could see something similar and learn from it?
I feel most comfortable with python but OK with other languages

**Comment by CrushedAsian255** [comment]
You can run direct SQL statements on the PostgreSQL database by running `docker exec -it immich_postgres psql -U postgres -d immich`, although I would not suggest doing so unless you have knowledge about Immich internals and/or SQL queries, as you can really screw things up if you do something bad. Also, good idea to [back up the database](https://immich.app/docs/administration/backup-and-restore/) (if you're not already doing so).
